apps/Welcome/components/yellowCommandLine/mainMenu.c

```diff
@@ -0,0 +1,121 @@
+//--------------------------------------------------------------------------------------------------
+/**
+ * Implementation of the main menu screen.
+ *
+ * Copyright (C) Sierra Wireless Inc.
+ */
+//--------------------------------------------------------------------------------------------------
+
+#include "cmdLine.h"
+#include "mainMenu.h"
+#include "ledScreen.h"
+#include "buzzerScreen.h"
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Draw the main menu.
+ */
+//--------------------------------------------------------------------------------------------------
+static void DrawMainMenu
+(
+    void
+)
+//--------------------------------------------------------------------------------------------------
+{
+    printf("\n= MAIN MENU =\n"
+           "\n"
+           "1. LEDs\n"
+           "2. Buzzer\n"
+           );
+}
+
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Handle a character of input while in the main menu.
+ */
+//--------------------------------------------------------------------------------------------------
+static void HandleMainMenuInput
+(
+    char input
+)
+//--------------------------------------------------------------------------------------------------
+{
+    switch (input)
+    {
+        case '1':
+
+            ledScreen_Enter();
+            break;
+
+        case '2':
+
+            buzzerScreen_Enter();
+            break;
+
+        default:
+            // Eat the invalid selection.
+            return;
+    }
+
+    cmdLine_Refresh();
+}
+
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Leave the main menu.
+ */
+//--------------------------------------------------------------------------------------------------
+static void Leave
+(
+    void
+)
+//--------------------------------------------------------------------------------------------------
+{
+    cmdLine_Exit();
+}
+
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * The Screen object for the main menu.
+ */
+//--------------------------------------------------------------------------------------------------
+static Screen_t MainMenu =
+{
+    drawFunc: DrawMainMenu,
```

Comment:
This type of initialization is GCC-only and isn't suggested for use given that there's a standard way to achieve the same thing. https://gcc.gnu.org/onlinedocs/gcc-9.1.0/gcc/Designated-Inits.html#Designated-Inits
`drawFunc: DrawMainMenu,` should be changed to `.drawFunc = DrawMainMenu,`

Reply:
Will do in next rev.